enhancements/cluster-logging/workload-identity-auth-for-azure-monitor-logs.md

```diff
@@ -0,0 +1,295 @@
+3. Use the access token in the log forwarding request to Azure Monitor Logs.
+
+The ClusterLogForwarder will:
+
+1. Conditionally project the service account token if workload identity authentication is configured.
+2. Create the collector configuration with the required fields for the Log Ingestion API, including either the path to the projected service account token or the client secret.
+
+### Proposed API
+
+#### New `AzureLogIngestion` output type
+
+Output
+
+```Go
+// New output type of AzureLogIngestion which can live side by side with the AzureMonitor output
+type AzureLogIngestion struct {
+  // The Data Collection Endpoint (DCE) or Data Collection Rule (DCR) logs ingestion endpoint URL.
+  //
+  // https://learn.microsoft.com/en-us/azure/azure-monitor/logs/logs-ingestion-api-overview#endpoint
+  //
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:XValidation:rule="isURL(self)", message="invalid URL"
+  // +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Url",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+  URL string `json:"url"`
+
+  // The Data Collection Rule's (DCR) Immutable ID
+  //
+  // A unique identifier for the data collection rule. This property and its value are automatically created when the DCR is created.
+  //
+  // https://learn.microsoft.com/en-us/azure/azure-monitor/data-collection/data-collection-rule-structure#properties
+  //
+  // +kubebuilder:validation:Required
+  // +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="DCR Immutable ID",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+  DcrImmutableId string `json:"dcrImmutableId"`
+
+  // The stream in the Data Collection Rule (DCR) that should handle the custom data
+  // 
+  // https://learn.microsoft.com/en-us/azure/azure-monitor/data-collection/data-collection-rule-structure#input-streams
+  //
+  // +kubebuilder:validation:Required
+  // +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Stream Name",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+  StreamName string `json:"streamName"`
+
+  // Authentication sets credentials for authenticating the requests.
+  //
+  // +kubebuilder:validation:Required
+  // +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Authentication Options",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+  Authentication *AzureLogIngestionAuth `json:"authentication"`
+
+  // Tuning specs tuning for the output
+  //
+  // +kubebuilder:validation:Optional
+  // +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Tuning Options"
+  Tuning *BaseOutputTuningSpec `json:"tuning,omitempty"`
+}
+```
+
+Authentication
+
+```Go
+type AzureLogIngestionAuth struct {
+  // ClientId points to the secret containing the client ID used for authentication.
+  // 
+  // This is the application ID that's assigned to your app. You can find this information in the portal where you registered your app.
```

Comment:
Stray comment?

Reply:
No this is the definition of the `ClientId`.